src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -1,29 +1,192 @@
+
+    Optional<ServerPoll> pollConcerned = pollRepository.findById(idPoll);
+    if (pollConcerned.isEmpty()) {
+      throw new ResourceNotFoundException();
+    } else if (!(pollTest.equals(pollConcerned.get()))) {
+      throw new WrongCredentialsException();
+    }
+
+    return repository.findById(maggieQ).orElseThrow(ResourceNotFoundException::new);
   }
 
-  @RequestMapping(value = "/question/{id}", method = RequestMethod.GET)
-  Question byId(@PathVariable QuestionIdentifier id) {
-    return repository.findById(id).orElseThrow(() -> new IllegalArgumentException("Not found."));
+  /**
+   * Insert a new {@link Question} in a {@link ServerPoll}.
+   *
+   * @param token       sender's token
+   * @param question    question to add
+   * @param idModerator moderator who should be the sender of the request
+   * @param idPoll      poll to add question in
+   * @return question added
+   * @throws ResourceNotFoundException if one parameter is broken
+   */
+  @PostMapping(value = "/mod/{idModerator}/poll/{idPoll}/question")
+  public Question insertQuestion(@RequestParam(name = "token") String token,
+      @RequestBody Question question,
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") ServerPollIdentifier idPoll)
+      throws ResourceNotFoundException, WrongCredentialsException {
+
+    testModeratorRight(idModerator, idPoll, token);
+
+    return pollRepository.findById(idPoll).get().newQuestion(repository, question);
+  }
+
+  /**
+   * Update a {@link Question} in a {@link ServerPoll}.
+   *
+   * @param token    sender's token
+   * @param question new version of the question
+   * @param idModo   moderator who should be the sender of the request
+   * @param idPoll   poll to add question in
+   * @param maggieQ  id of question to update
+   * @throws ResourceNotFoundException if one parameter is broken
+   */
+  @PutMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  public void updateQuestion(@RequestParam(name = "token") String token,
+      @RequestBody Question question,
+      @PathVariable(name = "idModerator") int idModo,
+      @PathVariable(name = "idPoll") ServerPollIdentifier idPoll,
+      @PathVariable(name = "idQuestion") QuestionIdentifier maggieQ)
+      throws ResourceNotFoundException, WrongCredentialsException {
+    testModeratorRight(idModo, idPoll, token);
+    ServerPoll pollTest = pollRepository.findById(idPoll).get();
+    Optional<Question> upQ = repository.findById(maggieQ);
+    if (upQ.isEmpty() || !(pollTest.equals(maggieQ.getIdxPoll()))) {
+      throw new ResourceNotFoundException();
+    }
+
+    upQ.get().setTitle(question.getTitle());
+    upQ.get().setDetails(question.getDetails());
+    upQ.get().setAnswersMax(question.getAnswersMax());
+    upQ.get().setAnswersMin(question.getAnswersMin());
+    upQ.get().setVisibility(question.getVisibility());
+    upQ.get().setIndexInPoll(question.getIndexInPoll());
```

Comment:
I'm not sure of how Hibernate works. Do we need to call `save` on the repository instance with the updated data ?

Reply:
Apparently not